src/Microsoft.OData.Edm/Schema/EdmContainedEntitySet.cs

```diff
@@ -12,7 +12,7 @@ namespace Microsoft.OData.Edm
     /// <summary>
     /// Represents an EDM contained entity set.
     /// </summary>
-    internal class EdmContainedEntitySet : EdmEntitySetBase, IEdmContainedEntitySet
+    public class EdmContainedEntitySet : EdmEntitySetBase, IEdmContainedEntitySet

```

Comment:
That's weird for a long time to make it internal. I do think the author wants to change it later so make it internal. So, Any reason to public it?

Reply:
A lot of stuff used to be unnecessarily `internal`, so I wouldn't be surprised if this was missed when things were cleaned up. I cannot see a reason for it to be internal, as it makes it impossible to create a `NavigationPropertySegment` that represents a contained navigation without reimplementing `EdmContainedEntitySet`.

In my code I ended up using a compiled lambda to create the instance I needed, but I'd much rather just use `new` :D

Reply:
Which constructor would you end up using if you didn't use the lambda?

Reply:
I'm using this one:

```
public EdmContainedEntitySet(IEdmNavigationSource parentNavigationSource, IEdmNavigationProperty navigationProperty)
```

Reply:
Thanks, I haven't take the time to fully evaluate this, but I would want to verify anything quality-related before taking this `public`. Just at a glance, for that constructor a `null` value for `navigationProperty` will result in a `NulReferenceException`, and it really should be an `ArgumentNullException` with the name of the parameter. Would you be willing to use your existing code for now and I will file an issue and make any quality updates that are necessary?

Reply:
Scope it down a bit and validated the input parameters for nullity - hopefully this will satisfy both of us :)